### PR TITLE
FR-15099 - Migrate to valtio

### PR DESCRIPTION
**Pull request by frontegg-david**

### CODE REVIEW

File: packages/vue/src/auth/mapAuthState.ts [hunk 1]
```diff
@@ -216,7 +216,7 @@ export const useFrontegg = () => {
 
   const loginWithRedirect = () => {
     const path = fronteggAuth.getCurrentRoute();
-    if (!path.startsWith(authState.routes.hostedLoginRedirectUrl ?? defaultFronteggRoutes.hostedLoginRedirectUrl)) {
+    if (!path.startsWith(authState.routes.hostedLoginRedirectUrl ?? defaultFronteggRoutes.hostedLoginRedirectUrl ?? "oauth/callback")) {
```
Comment (doregg) [inline packages/vue/src/auth/mapAuthState.ts:219]: what's this?

File: packages/vue/src/auth/service.ts [hunk 2]
```diff
@@ -16,41 +14,20 @@ import {
...
-  return reducerActions.reduce((p, n) => ({ ...p, ...n }), {}) as CaseReducerActions<T>;
-};
-
 export class FronteggAuthService implements FronteggPluginService {
   pluginConfig!: PluginConfig;
   router?: VueRouter | null;
   private readonly _routes!: AuthPageRoutes;
-  private store?: EnhancedStore<FronteggStore>;
-  private state: AuthState = authInitialState;
+  private store!: FronteggStore;
```
Comment (doregg) [inline packages/vue/src/auth/service.ts:29]: what's the '!'?

File: packages/vue/src/auth/service.ts [hunk 3]
```diff
@@ -16,41 +14,20 @@ import {
...
-  const reducerActions = reducerKeys.map(key => ({ [key]: (authActions as any)[key as keyof AuthActions] }));
-  return reducerActions.reduce((p, n) => ({ ...p, ...n }), {}) as CaseReducerActions<T>;
-};
-
 export class FronteggAuthService implements FronteggPluginService {
   pluginConfig!: PluginConfig;
   router?: VueRouter | null;
   private readonly _routes!: AuthPageRoutes;
-  private store?: EnhancedStore<FronteggStore>;
-  private state: AuthState = authInitialState;
```
Comment (doregg) [inline packages/vue/src/auth/service.ts:53]: why?

File: packages/vue/src/auth/service.ts [hunk 4]
```diff
@@ -67,7 +44,7 @@ export class FronteggAuthService implements FronteggPluginService {
 
   constructor({ router, ...options }: AuthPluginOptions) {
     this._routes = {
-      ...authInitialState.routes,
+      ...defaultFronteggRoutes,
```
Comment (doregg) [inline packages/vue/src/auth/service.ts:47]: why?

File: packages/vue/src/auth/service.ts (hunk 2 above)
Comment (frontegg-david) [reply packages/vue/src/auth/service.ts:29]: late initialized non-nullable value

File: packages/vue/src/auth/service.ts (hunk 4 above)
Comment (frontegg-david) [reply packages/vue/src/auth/service.ts:47]: moved default routes to vue

File: packages/vue/src/auth/service.ts (hunk 3 above)
Comment (frontegg-david) [reply packages/vue/src/auth/service.ts:53]: no need it anymore

File: packages/vue/src/auth/mapAuthState.ts (hunk 1 above)
Comment (frontegg-david) [reply packages/vue/src/auth/mapAuthState.ts:219]: this is the default route, fix lint of nullable check

**Review by TomerFrontegg — APPROVED** [review]

**Pull request merged**